src/instrument/mongoose_instrument_prometheus.erl

```diff
@@ -54,19 +82,26 @@ handle_metric_event(EventName, LabelValues, MetricName, MetricType, Measurements
             ok
     end.
 
+-spec metric_help(mongoose_instrument:event_name(), mongoose_instrument:metric_name()) -> help().
 metric_help(EventName, MetricName) ->
     lists:flatten(io_lib:format("Event: ~p, Metric: ~p", [EventName, MetricName])).
 
+-spec full_metric_name(mongoose_instrument:event_name(), mongoose_instrument:metric_name()) ->
+          name().
 full_metric_name(EventName, MetricName) ->
-    list_to_atom(atom_to_list(EventName) ++ "_" ++ atom_to_list(MetricName)).
+    atom_to_list(EventName) ++ "_" ++ atom_to_list(MetricName).
```

Comment:
This in comparison to the previous code is pretty much having the same performance, true, but, here on every metric update we're concatenating the same two lists again and again. Is this necessary? Maybe the metric name can be `[EventName, MetricName]`? Or some similar structure that implies no copying? Just asking out loud, I don't know the required API for the prometheus library here.

Reply:
Prometheus ilbrary accepts only atoms or flat strings. Any changes would require modifying `prometheus.erl`.

Reply:
Oh, so unfortunate 🤔